fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -222,9 +222,10 @@ public CommandProcessingResult submitApplication(final JsonCommand command) {
             boolean isMifosChannel = false;
             Long mifosChannelId = null;
             String mifosChannelHash = null;
+            String mifosChannel = "Mifos";
             List<Channel> channels = loanProduct.getRepaymentChannels();
             for (Channel channel : channels) {
-                if (channel.getName().equals("Mifos")) {
+                if (channel.getName().toUpperCase().equals(mifosChannel.toUpperCase())) {
```

Comment:
why not use equals ignorecase and invert to allow mifoschannel come first since you are sure it's not going to be null